ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/DiskPolicy.hs

```diff
@@ -79,16 +85,25 @@ data DiskPolicy = DiskPolicy {
 -- take a snapshot roughly every @k@ blocks. It does mean the possibility of
 -- an extra unnecessary snapshot during syncing (if the node is restarted), but
 -- that is not a big deal.
-defaultDiskPolicy :: SecurityParam -> DiskPolicy
-defaultDiskPolicy (SecurityParam k) = DiskPolicy {..}
+defaultDiskPolicy :: SecurityParam -> RequestedInterval -> DiskPolicy
```

Comment:
Would you update the Haddock on this to mention the disjunction in the logic of when we take snapshots? EG I'm not seeing anything that covers the `k*2` case. So I think adding an mention of the new5`RequestedInterval` would be an improvement.

Reply:
I've updated the comments but I must admit that I dislike them because they do the following:
1. state the obvious what can be clearly seen in the code
example: 
```
"- We want to take a snapshot every 50k blocks or roughly every requested interval."
```

2. comment the usage of the function (as if the function knew exactly how it is being used)
example
```
      -- * The distance in terms of blocks applied to the /oldest/ ledger
      --   snapshot in memory. During normal operation, this is the number of
      --   blocks written to the ImmutableDB since the last snapshot. On
      --   startup, it is computed by counting how many immutable blocks we had
      --   to reapply to get to the chain tip. This is useful, as it allows the
```

Reply:
Also the previous comment mentioned following

```
--- (...) when @k = 2160@ (for other values of @k@ we scale
--- proportionally), but not more rapidly than 10 per hour (which is important
--- for bulk sync).
```

No such thing was reflected in the code so I removed the comment. But I'm stating it here explicitly, maybe I'm missing something

Reply:
One last remark the comments: if we treat comments as "double bookkeeping" for the logic of our code, then those are not the best tool for the job as they can easily get out of sync and it requires the bookkeeper (human.closely looking at each PR) to keep them at sync. There is a better tool for the job and it is called specs/tests. Tests govern that the change in one breaks the other. And if it is the documentation that we are missing, then we can easily generate it from the spec. My #2 #cents :)

Reply:
We chatted on our call. We agreed that the comments especially on `defaultDiskPolicy` could have less implementation details. Something like "a disk policy suitable for use with `cardano-node`". The magic number explanations etc belong within the definition as inline comments.